### ISSUE TITLE
Undefined array key error when using Imgix transformer with a single image dimension

### ISSUE THREAD

**Issue by martyspain** [spacecatninja/craft-imager-x#173]
### I'm submitting a... 

- [x] bug report
- [ ] feature request
- [ ] general question

### Steps to reproduce

1. Request an image transform with Imager X using the Imgix transformer
2. In the request, only specify one dimension for the image (width OR height) but not both
3. Observe an ErrorException is thrown

### Description
With the change made to `src/transformers/ImgixTransformer.php` in 4.1.4, the Imgix transformer now requires both width and height to be specified in order to work, whereas prior to this change you could specify just width _or_ height and the image would be transformed accordingly depending on your `mode` settings. Now, if you don't specify both params, the image transform will fail with an `ErrorException: Undefined array key "h"` error (or `"w"` if you specify height but not width).

For example, this code will generate the exception:

```
{% set image = field.image.one() %}
{% set imageSizes = [
    {
        width: 450
    },
    {
        width: 700
    }
] %}
{% set transform = craft.imager.transformImage(image, imageSizes,
    {
        auto: 'compress,format',
        allowUpscale: false,
        mode: 'crop',
        transformerParams: {
            usm: 20
        }
    }
) %}
```

### Displayed errors, stack trace, relevant logs  

```
PHP Warning – [yii\base\ErrorException](http://www.yiiframework.com/doc-2.0/yii-base-errorexception.html)
Undefined array key "h"
1. in /var/www/html/vendor/spacecatninja/imager-x/src/transformers/ImgixTransformer.phpat line 233
 
        // Assume that the reset of the values left in the transform object is Imgix specific
        foreach ($transform as $key => $val) {
            $r[$key] = $val;
        }
 
        // If allowUpscale is disabled, use max-w/-h instead of w/h
        if (isset($r['fit']) && !$config->getSetting('allowUpscale', $transform)) {
            if ($r['fit'] === 'crop' && $image instanceof Asset && ($r['w'] > $image->getWidth() || $r['h'] > $image->getHeight())) {
                $sourceRatio = $image->getHeight() / $image->getWidth();
                $transformRatio = (int)$r['h'] / (int)$r['w'];
 
                if ($sourceRatio > $transformRatio) {
                    $r['w'] = $image->getWidth();
                    $r['h'] = $image->getWidth() * $transformRatio;
                } else {
                    $r['h'] = $image->getHeight();
                    $r['w'] = $image->getHeight() / $transformRatio;
```

### Additional info

- Imager version & edition: 4.1.4
- Imager transformer: Imgix
- Craft version: 4.2.5.1
- PHP version: 8.0
- Image driver & version: Imagick 3.6.0 (ImageMagick 6.9.11-60)

**Comment by aelvan** [comment]
Thanks for reporting, will fix shortly. Although, the reason it fails is because `mode`/`fit` is `crop`, while only one dimension is given. The fix will be to fall back to changing `fit` to `clip` in these cases.

**Comment by aelvan** [comment]
Fixed in 4.1.5.
